packages/graphback-runtime/src/service/DefaultCRUDService.ts

```diff
@@ -26,99 +31,100 @@ export class DefaultCRUDService<T = any, GraphbackContext = any>
         this.logger = logger || defaultLogger;
     }
 
-    public async create(inputType: InputModelTypeContext, data: T, context?: GraphbackContext): Promise<T> {
-        this.logger.log(`Creating object ${inputType.name}`)
-        const result = await this.db.create(inputType, data, context);
-        if (this.pubSub && inputType.config.subCreate) {
-            const topic = subscriptionTopicMapping(GraphbackOperationType.CREATE, inputType.name);
-            const payload = this.buildEventPayload('new', inputType, result);
+    public async create(name: string, data: T, context?: GraphbackContext): Promise<T> {
+        this.logger.log(`Creating object ${name}`)
+        const result = await this.db.create(name, data, context);
+        // TODO subscriptions
+        if (this.pubSub && context && context.publish) {
+            const topic = subscriptionTopicMapping(GraphbackOperationType.CREATE, name);
+            const payload = this.buildEventPayload('new', name, result);
             await this.pubSub.publish(topic, payload);
         }
 
         return result;
     }
-    public async update(inputType: InputModelTypeContext, id: string, data: T, context?: GraphbackContext): Promise<T> {
-        this.logger.log(`Updating object ${inputType.name}`)
+    public async update(name: string, id: string, data: T, context?: GraphbackContext): Promise<T> {
+        this.logger.log(`Updating object ${name}`)
 
-        const result = await this.db.update(inputType, id, data, context);
-        if (this.pubSub && inputType.config.subUpdate) {
-            const topic = subscriptionTopicMapping(GraphbackOperationType.UPDATE, inputType.name);
-            const payload = this.buildEventPayload('updated', inputType, result);
+        const result = await this.db.update(name, id, data, context);
+        if (this.pubSub && context && context.publish) {
+            const topic = subscriptionTopicMapping(GraphbackOperationType.UPDATE, name);
+            const payload = this.buildEventPayload('updated', name, result);
             await this.pubSub.publish(topic, payload);
         }
 
         return result;
     }
 
     // tslint:disable-next-line: no-reserved-keywords
-    public async delete(inputType: InputModelTypeContext, id: string, context?: GraphbackContext): Promise<string> {
-        this.logger.log(`deleting object ${inputType.name}`)
+    public async delete(name: string, id: string, context?: GraphbackContext): Promise<string> {
+        this.logger.log(`deleting object ${name}`)
 
-        const result = await this.db.delete(inputType, id, context);
-        if (this.pubSub && inputType.config.subDelete) {
-            const topic = subscriptionTopicMapping(GraphbackOperationType.DELETE, inputType.name);
-            const payload = this.buildEventPayload('deleted', inputType, result);
+        const result = await this.db.delete(name, id, context);
+        if (this.pubSub && context && context.publish) {
```

Comment:
What would happen if subscriptions are enabled in `graphback.json`, for example, `subDelete: true`, but `publish` is false in `GraphbackContext`? The subscriptions would be added to the GraphQL schema but would not work update?

It is a question more than anything.

Reply:
Subscriptions are not implemented yet. Working on this as we speak

Reply:
Generally, the idea will be that we will need to pass info somehow about subscription being enabled in config.